javascript/ql/src/Statements/UselessConditional.ql

```diff
@@ -120,9 +120,12 @@ predicate whitelist(Expr e) {
  */
 predicate isConditional(ASTNode cond, Expr e) {
   e = cond.(IfStmt).getCondition() or
+  e = cond.(WhileStmt).getExpr() or
+  e = cond.(ForStmt).getTest() or
```

Comment:
How about just generalising all the way to `LoopStmt.getTest()`? Or do we not want this to cover `DoWhileStmt.getTest()`?

Reply:
Actually it looks like I got that backwards. `EphemeralLoop.ql` whitelists Emscripten-generated files because of `do {} while(0)`, but this query already whitelists constants, so it should be a problem.

Reply:
Updated to use `LoopStmt`